### PR TITLE
feat: password reset email

### PR DESCRIPTION
**Pull request by Skibo555**
The endpoint allows users to reset their password by providing the password they used in registration.
​

## Description

The endpoint takes the email the user provided and compare it again the record in the database if it matches any email, if it does, it sends a password reset link into the user's email and returns 200 OK status code. If the email does not exist in the database, it returns "We don't have user with the provided email in our database." and 404 status code.
​

## Related Issue (Link to issue ticket)
https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/51

​

## Motivation and Context

This will allow users to retrieve their email should in case anything happen to their account provided thy know the email address.
​

## How Has This Been Tested?
Yes, I tested it using pytest and Postman.

​

## Screenshots (if appropriate - Postman, etc):

​
Screenshot 2024-07-24 at 15 10 24
Screenshot 2024-07-24 at 15 15 21
Screenshot 2024-07-24 at 15 15 58
Screenshot 2024-07-24 at 15 53 11
Screenshot 2024-07-24 at 19 20 26

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — APPROVED** [review]

**Comment by Skibo555** [comment]
I've been deactivated from the slack channel though!

File: api/v1/routes/auth.py [hunk 1]
```diff
@@ -173,10 +183,66 @@ async def verify_signin_token(token_schema: TokenRequest, db: Session = Depends(
...
     )
-    
+
 
 # Protected route example: test route
 @auth.get("/admin")
 def read_admin_data(current_admin: Annotated[User, Depends(user_service.get_current_super_admin)]):
     return {"message": "Hello, admin!"}
 
+def generate_access_token(data: dict, expires_delta: Optional[timedelta] = None):
```
Comment (joboy-dev) [inline api/v1/routes/auth.py:193]: From here, all of this needs to be in user service. 
Generate access token has been taken care of

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Remove this .idea from here
Don't check it ingo github

File: api/v1/routes/auth.py (hunk 1 above)
Comment (Skibo555) [reply api/v1/routes/auth.py:193]: Once it's put there, it does not work again for some reasons I can't explain, so I had to leave the functions in the auth file.

**Comment by joboy-dev** [comment]
Closing this PR because this endpoint has been implemented and merged

**Pull request closed without merging**